### DISCUSSION TITLE
Arduino to Raspberry Pi Communication

### DISCUSSION THREAD

**Discussion by andymk3** [gfunkbus76/Digifiz-Dash#3] [Q&A] [answered]
Hi, 

Just wondering how you are connecting the Arduino to the Raspberry Pi? Are you connecting them via USB Serial? What's the setup between the 2 to communicate? 

Great project!

**Comment by gfunkbus76** [comment, marked as answer]
Hey Andy,

My plan has me connecting to my receiving Arduino(end of my canbus line) via USB serial.

I'll have the Arduino plugged into my key on power.

Currently I've looked into getting node red to read the serial information line by line and I'll send that to the appropriate mqtt topic. I want to clean up my Arduino code and then start reading and interpreting it on the node red system. I'll probably average the values out to smooth them a bit. I'll have to play with it.

**Reply by andymk3** [reply]
Ah okay sounds good! 

I've been working on altering your code my mine a lot recently, got RPM gauge working today (analogue style), so happy with that!

**Reply by gfunkbus76** [reply]
Nice! That's awesome. I had found a video on YouTube ages ago, and the guy shared his GitHub recently. Similar lol but cleaner code structure... It's called Miata-dash.

Trying to see how to better set up my code with classes and separating files from the main to avoid such a huge code block lol. Working on incorporating his general structure. 

I'm new to Python and programming in general so it's all a learning curve lol.

**Reply by andymk3** [reply]
I just checked that out, that really is nice and clean, I think I will have to do the same! 

I'm fairly new to Python too, it's slowly starting to make sense. Also learning to use Adobe XD and Illustrator at the same time! 

A quick vid of my rev counter: https://www.youtube.com/watch?v=A03EsbD9WPM
Not a final design, just getting to grips with it.

**Reply by gfunkbus76** [reply]
Nice! Good progress, I like your slider bar for testing the node-red ... I'll have to do that! Looks easier lol. I've updated my git with some updates. Integrating some of what miata-dash has in their setup. I'm going to keep trying to split the main.py into easier modules ... Cleaned up some of the MQTT stuff also. Would like to get my mqtt stuff into a separate .py ... Working on it anyways.

My progress has slowed as my main dev computer has been overtaken by my wife while she does some online courses :\... Lookinfg forward to seeing what you make with your dash!

Cheers!
